### PR TITLE
Sidebar and Header styled components

### PR DESCRIPTION
**Pull request by enwong10**
### Describe your changes 
Reusable styled components for sidebar items

Migrating to styled components allows for the reuse of styles on both a codebase level and a component level. As evident among the sidebar item suite (`Top Pools`, `Recent Pools`, `Favorite Pools`, etc.), styles specific to the component can be shared. Additionally, common styles such as a generic `FlexContainer`, `GridContainer`, and even `Status` (for Ranges) can be reused in other parts of the codebase. Specific component styles also extend common classes (for example, `HeaderGrid` is a `GridContainer` with extra (or overriding) styling).  

### Link the related issue
_Closes #2704_

### Checklist before requesting a review
- [x] Is this PR ready for merge? (Please convert to a draft PR otherwise)
- [x] I have performed a self-review of my code.
- [x] Did I request feedback from a team member prior to merge? 
- [x] Does my code following the style guide at `docs/CODING-STYLE.md`?

### CODE REVIEW

**Comment by enwong10** [comment]
Closed search icon bottom border missing 
Search string persist when closing 
Arrows when opening and closing sections

File: src/styled/Common.ts [hunk 1]
```diff
@@ -66,10 +40,115 @@ const Color = css<{
...
+    fullHeight?: boolean;
+    fullWidth?: boolean;
+    flexDirection?: 'row' | 'column';
+    justifyContent?: string;
+    alignItems?: string;
+    overflow?: string;
+}
+const Flex = css<FlexProps>`
+    display: flex;
+    ${({ gap }) => gap && `gap: ${gap}px`};
```
Comment (cadezacharias-hype) [inline src/styled/Common.ts:115]: I think we should allow gap here to be any denomination - I see it sometimes listed in `rem`

File: src/styled/Common.ts [hunk 2]
```diff
@@ -66,10 +40,115 @@ const Color = css<{
...
+interface FlexProps {
+    gap?: number;
+    fullHeight?: boolean;
+    fullWidth?: boolean;
+    flexDirection?: 'row' | 'column';
+    justifyContent?: string;
+    alignItems?: string;
+    overflow?: string;
+}
+const Flex = css<FlexProps>`
```
Comment (cadezacharias-hype) [inline src/styled/Common.ts:113]: What do you think of rewriting this likeso for readability?

```
const Flex = css<FlexProps>`
  ${({ gap, fullWidth, fullHeight, flexDirection, justifyContent, alignItems, overflow }) => {
    return `
      display: flex;
     ${gap && `gap: ${gap}px;`}
     ${fullWidth && 'width: 100%;'}
     ...
    `;
  }}
`;
```

Comment (cadezacharias-hype) [reply src/styled/Common.ts:113]: I noticed I was struggling to find what would be the defaults at a glance just bc of the noise

**Comment by enwong10** [comment]
Use constants for media breakpoints and hardcoded consts

File: src/App/components/Sidebar/SidebarSearchResults/OrdersSearchResults.tsx [hunk 3]
```diff
@@ -85,16 +94,52 @@ export default function OrdersSearchResults(props: propsIF) {
...
+            </Text>
             {searchedLimitOrders.length ? (
-                <>
-                    <header className={styles.header}>
-                        <div>Pool</div>
-                        <div>Price</div>
-                        <div>Value</div>
-                    </header>
-                    <ol className={styles.main_result_container}>
+                <FlexContainer flexDirection='column' fullWidth>
```
Comment (cadezacharias-hype) [inline src/App/components/Sidebar/SidebarSearchResults/OrdersSearchResults.tsx:107]: So this is something I don't love - we managed to turn 6 lines into like 35 lines.  

Seems like a weird thing bc this is the whole reason you'd use classes. You don't want to do the same inline styling for 3 of the same components. As a dev I'd much rather just create the text component I want and then uses it 3 times. or however many is needed.

File: src/App/components/Sidebar/SidebarSearchResults/PoolsSearchResults/PoolsSearchResults.tsx [hunk 4]
```diff
@@ -52,16 +57,52 @@ export default function PoolsSearchResults(props: propsIF) {
...
     return (
-        <div>
-            <h4 className={styles.card_title}>Pools</h4>
+        <FlexContainer
+            flexDirection='column'
+            justifyContent='center'
+            alignItems='flex-start'
+            gap={8}
+        >
+            <Text fontWeight='500' fontSize='body' color='accent5'>
```
Comment (cadezacharias-hype) [inline src/App/components/Sidebar/SidebarSearchResults/PoolsSearchResults/PoolsSearchResults.tsx:66]: Might also be nice to have constants for the different weights we use.  Or at least in the Text component have a subset of weights available. Just so I don't type in 450 and suddenly we're using slightly different weights throughout.

File: src/App/components/Sidebar/SidebarSearchResults/PoolsSearchResults/PoolsSearchResults.tsx [hunk 5]
```diff
@@ -52,16 +57,52 @@ export default function PoolsSearchResults(props: propsIF) {
...
             {searchedPools.length ? (
-                <>
-                    <header className={styles.header}>
-                        <div>Pool</div>
-                        <div>Volume</div>
-                        <div>TVL</div>
-                    </header>
-                    <ol className={styles.main_result_container}>
+                <FlexContainer flexDirection='column' fullWidth>
+                    <GridContainer
```
Comment (cadezacharias-hype) [inline src/App/components/Sidebar/SidebarSearchResults/PoolsSearchResults/PoolsSearchResults.tsx:71]: Another instance of 3 of the same component getting expanded.

File: src/App/components/Sidebar/SidebarSearchResults/PositionsSearchResults.tsx [hunk 6]
```diff
@@ -94,16 +103,52 @@ export default function PositionsSearchResults(props: propsIF) {
...
+                        numCols={3}
+                        fullWidth
+                        fontWeight='300'
+                        fontSize='body'
+                        color='text2'
+                        style={{ borderBottom: '1px solid var(--dark3)' }}
+                        padding='0 0 4px 0'
+                    >
+                        <Text
+                            fontWeight='300'
```
Comment (cadezacharias-hype) [inline src/App/components/Sidebar/SidebarSearchResults/PositionsSearchResults.tsx:127]: Same thing again - this style is reused enough to where we should be able to make it a common component.

File: src/App/components/Sidebar/SidebarSearchResults/TxSearchResults/TxLI.tsx [hunk 7]
```diff
@@ -22,17 +22,20 @@ export default function TxLI(props: propsIF) {
...
+            padding='4px'
+            onClick={() => handleClick(tx)}
+        >
             <p>
                 {tx.baseSymbol} / {tx.quoteSymbol}
             </p>
             <p style={{ textAlign: 'center' }}>{txType}</p>
-            <p
-                style={{ textAlign: 'center' }}
-                className={styles.status_display}
```
Comment (cadezacharias-hype) [inline src/App/components/Sidebar/SidebarSearchResults/TxSearchResults/TxLI.tsx:32]: confirming that this is a dead class - I don't see it removed anywhere.

File: src/App/components/Sidebar/SidebarSearchResults/TxSearchResults/TxSearchResults.tsx [hunk 8]
```diff
@@ -53,16 +58,52 @@ export default function TxSearchResults(props: propsIF) {
...
+                        numCols={3}
+                        fullWidth
+                        fontWeight='300'
+                        fontSize='body'
+                        color='text2'
+                        style={{ borderBottom: '1px solid var(--dark3)' }}
+                        padding='0 0 4px 0'
+                    >
+                        <Text
+                            fontWeight='300'
```
Comment (cadezacharias-hype) [inline src/App/components/Sidebar/SidebarSearchResults/TxSearchResults/TxSearchResults.tsx:82]: 👀

File: src/components/Chat/MessagePanel/Room/Room.module.css [hunk 9]
```diff
@@ -125,14 +125,14 @@
...
-    color: rgba(235, 235, 255, 0.4);
+    color: var(--text2);
     border-radius: 4px;
 }
 
 .star_icon_selected_room {
     height: 20px;
     width: 20px;
-    color: rgba(235, 235, 255, 0.4);
+    color: var(--text2);
```
Comment (cadezacharias-hype) [inline src/components/Chat/MessagePanel/Room/Room.module.css:135]: Nice use of vars

File: src/components/Global/Sidebar/FavoritePools.tsx [hunk 10]
```diff
@@ -35,14 +34,18 @@ export default function FavoritePools(props: propsIF) {
...
     return (
-        <SidebarPoolsListContainer fontSize={'body'}>
-            <SidebarPoolsListHeaderContainer>
-                <SidebarPoolsListHeader>Pool</SidebarPoolsListHeader>
-                <SidebarPoolsListHeader>Volume</SidebarPoolsListHeader>
-                <SidebarPoolsListHeader>TVL</SidebarPoolsListHeader>
-            </SidebarPoolsListHeaderContainer>
+        <FlexContainer flexDirection='column' fontSize='body' fullHeight>
+            <HeaderGrid numCols={3} color='text2' padding='4px 0'>
+                {['Pool', 'Volume', 'TVL'].map((item) => (
```
Comment (cadezacharias-hype) [inline src/components/Global/Sidebar/FavoritePools.tsx:39]: I like the idea - I think personal preference for readability in these situations would be just to have 3 divs listed out.  But it's totally just a preferences. 

10 Items for sure would map, 2 items for should would not.  The grey area is in between.

File: src/components/Global/Sidebar/PoolsListItem.tsx [hunk 11]
```diff
@@ -77,18 +77,28 @@ export default function PoolsListItem(props: propsIF) {
...
+            numCols={3}
+            color='text2'
         >
-            <SidebarPoolsListItemColumn>
-                {pool.base.symbol} / {pool.quote.symbol}
-            </SidebarPoolsListItemColumn>
-            <SidebarPoolsListItemColumn>{volume}</SidebarPoolsListItemColumn>
-            <SidebarPoolsListItemColumn>{tvl}</SidebarPoolsListItemColumn>
-        </SidebarPoolsListItemContainer>
+            {[`${pool.base.symbol} / ${pool.quote.symbol}`, volume, tvl].map(
```
Comment (cadezacharias-hype) [inline src/components/Global/Sidebar/PoolsListItem.tsx:90]: similar scenario to above

File: src/components/Trade/TradeTabs/Ranges/RangeCardHeader.module.css [hunk 12]
```diff
@@ -3,7 +3,7 @@
     /* flex-flow: row nowrap; */
     align-items: center;
     padding: 2px 8px;
-    color: rgba(235, 235, 255, 0.4);
+    color: var(--text2);
```
Comment (cadezacharias-hype) [inline src/components/Trade/TradeTabs/Ranges/RangeCardHeader.module.css:6]: Q: are we still uses these css modules?  Will they get converted to styled components?

**Review by cadezacharias-hype — CHANGES_REQUESTED** [review]
Main complaint is a few places where we've expanded divs to text components with a ton of the same inline styling.  

The rest of the comments are mainly philosophical points we can discuss if need be.

File: src/App/components/Sidebar/SidebarSearchResults/PoolsSearchResults/PoolsSearchResults.tsx (hunk 4 above)
Comment (enwong10) [reply src/App/components/Sidebar/SidebarSearchResults/PoolsSearchResults/PoolsSearchResults.tsx:66]: will be done in Phase 2

**Review by benwolski — APPROVED** [review]
looks great!

**Pull request merged**